packages/playwright/src/matchers/expect.ts

```diff
@@ -129,7 +128,20 @@ function createExpect(info: ExpectMetaInfo) {
 
       if (property === 'extend') {
         return (matchers: any) => {
-          expectLibrary.extend(matchers);
+          const wrappedMatchers: any = {};
+          Object.entries(matchers).forEach(([name, matcher]) => {
+            wrappedMatchers[name] = function(...args: any[]) {
+              const { isNot, promise, utils } = this;
+              const newThis: ExpectMatcherState = {
+                isNot,
+                promise,
+                utils,
+                timeout: currentExpectTimeout({})
```

Comment:
Might as well move `currentExpectTimeout` and `setCurrentExpectConfigureTimeout` into this file from the `globals.ts` and turn them local variables here.